src/math/Euler.js

```diff
@@ -316,9 +316,38 @@ class Euler {
 
 	}
 
-}
+	// r148
+
+	static get RotationOrders() {
+
+		console.error( 'THREE.Euler.RotationOrders has been removed.' );
+
+	}
+
+	static set RotationOrders( value ) {
+
+		console.error( 'THREE.Euler.RotationOrders has been removed.' );
+
+	}
+
+	// r148
 
-Euler.DefaultOrder = 'XYZ';
-Euler.RotationOrders = [ 'XYZ', 'YZX', 'ZXY', 'XZY', 'YXZ', 'ZYX' ];
+	static get DefaultOrder() {
+
+		console.warn( 'THREE.Euler: Euler.DefaultOrder has been renamed to Euler.DEFAULT_ORDER.' );
+		return Euler.DEFAULT_ORDER;
+
+	}
+
+	static set DefaultOrder( value ) {
+
+		console.warn( 'THREE.Euler: Euler.DefaultOrder has been renamed to Euler.DEFAULT_ORDER.' );
+		Euler.DEFAULT_ORDER = value;
+
+	}
+
+	static DEFAULT_ORDER = 'XYZ';
```

Comment:
I'm afraid we cannot use static class fields in core yet... They are still not very much supported, if I remember correctly (the last try of class fields was to use public class fields for `isClass` properties).

Reply:
What is the acceptable pattern to use to provide a default warning for the deprecated static properties?

Reply:
What is the problem, exactly? https://caniuse.com/mdn-javascript_classes_static_class_fields

Reply:
The discussion was here: https://github.com/mrdoob/three.js/pull/24237

Reply:
Let's avoid the usage of `static` otherwise the PR will be blocked.

I suggest to just remove the old syntax without a fallback. A note a migration guide is for this case sufficient.

Reply:
@LeviPesin said
>They are still not very much supported

 Based on what, please.

Reply:
Like mentioned in https://github.com/mrdoob/three.js/pull/24237 certain build tools still have issues with class properties.

Reply:
The way I read #24237, everyone else seems to prefer moving forward.

This is why we have a dev branch. If a PR causes problems with a bundler, we can revert it.

Reply:
The problem is that bundler issues mostly appear in production.

Reply:
We have plenty of smart contributors who can test it.

Reply:
I did the tests

Reply:
I'll report here as well, I did the tests about what bundler this PR works on. Most of the concern is about Webpack 4.

I previded the data, now you just have to decide on a timeline to start using class fields.

| Bundler | Current version | Works since version | Works since date |
|-------|-------|---------|-----------------|
| Webpack | `v5.75.0` | `v5.0.0` | Oct 2020 |
| create-react-app | `v5.0.1` | `v5.0.0` | Dec 2021 |
| Next.js | `v13.0.6` | `v10.0.0` | Oct 2020 |
| Rollup | `v3.7.4` | `v2.4.0` | Apr 2020 |
| esbuild | `v0.16.7` | `v0.8.13` | Nov 2020 |
| Vite | `v4.0.1` | `v2.0.0` | Feb 2021 |

Reply:
We have decided in https://github.com/mrdoob/three.js/pull/25168#issuecomment-1362393766 to stick to ECMA2018. Static class fields are ES2022. So please revert the usage otherwise the PR can't be merged.

Reply:
> What is the acceptable pattern to use to provide a default warning for the deprecated static properties?

Maybe we can use `Object.defineProperties( Euler, ... )`?

Reply:
Maybe you can suggest a solution you have (1) tested and (2) works.

Reply:
Perhaps you can suggest a solution you have tested.